### PR TITLE
1629 custom auth strategy adobe campaign

### LINKED ISSUE #1629: Custom auth strategy for Adobe Campaign
### Is your feature request related to a specific problem?

The Adobe Campaign connector currently uses the `bearer` authentication strategy with a temporary token that is generated manually. This ticket outlines an automated way to refresh this token so a manual refresh is not required

### Describe the solution you'd like

This is an outline of the steps required

1. Generate a JWT with the following values
```
{
    "exp": <current epoch time + 60 seconds>,
    "iss": "<organization_id>",
    "sub": "<technical_account_id>",
    "https://ims-na1.adobelogin.com/s/ent_campaign_sdk": true,
    "aud": "https://ims-na1.adobelogin.com/c/<client_id>"
}
```
2. Sign the JWT with the `private_key`
3. Post the signed JWT to `https://ims-na1.adobelogin.com/ims/exchange/jwt`
```
client_id={api_key_value}&client_secret={client_secret_value}&jwt_token={base64_encoded_JWT}
```
4. Use the returned `access_token` in every subsequent request until the time in the `expires_in` value has elapsed.

Complete instructions can be found here https://developer.adobe.com/developer-console/docs/guides/authentication/JWT/#using-jwt-libraries-and-creation-tools

### PR DESCRIPTION
**Pull request by galvana**
Closes #1629 

### Code Changes

* [x] Added new `adobe_campaign` authentication strategy that does the following
  * [x] Generates and signs a short-lived JWT
  * [x] Exchanges the JWT for an access token with a 24 hour TTL
  * [x] Refreshes the token after 24 hours have elapsed
* [x] Updates the adobe_campaign_config.yml with the new required connector params
  * [x] Adding descriptions to help with configuration

### Steps to Confirm

* [x] Re-ran the connection test successfully

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation Updated:
  * [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  * [ ] documentation issue created (tag docs-team to complete issue separately)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`

### CODE REVIEW

File: data/saas/config/adobe_campaign_config.yml [hunk 1]
```diff
@@ -3,27 +3,39 @@ saas_config:
...
   type: adobe_campaign
   description: A schema representing the Adobe Campaign connector for Fidesops
-  version: 0.0.1
+  version: 0.0.2
 
   connector_params:
     - name: domain
       default_value: mc.adobe.io
+    - name: organization
+      description: e.g. ORGANIZATION for production or ORGANIZATION-mkt-stage for staging
```
Comment (galvana) [inline data/saas/config/adobe_campaign_config.yml:12]: Added some descriptions to help with the configuration.

File: data/saas/config/adobe_campaign_config.yml [hunk 2]
```diff
@@ -3,27 +3,39 @@ saas_config:
...
+    - name: client_secret
+    - name: technical_account_id
+      description: e.g. ***@techacct.adobe.com
+    - name: private_key
+      description: PEM formatted
     - name: namespace
       default_value: defaultNamespace1
       description: The namespace to use for data protections requests
     - name: regulation
+      options: [CCPA, GDPR]
```
Comment (galvana) [inline data/saas/config/adobe_campaign_config.yml:25]: Enforcement of expected values.

**Review by adamsachs — APPROVED** [review]
@galvana this looks good to me - cleanly done. thanks for jumping on it quickly!

only question is whether you've been able to run the tests manually/locally to confirm things work as expected, given it seems we're (understandably) skipping the adobe tests as part of CI?

**Comment by galvana** [comment]
@adamsachs, yes, I ran a local test to confirm the connectivity worked as part of a privacy request.

**Pull request merged**